File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string): string =
    logLine.Split(' ', 2,
                  System.StringSplitOptions.TrimEntries)[1]
    
let logLevel(logLine: string): string =
    logLine.Split(' ', 2)[0]
    |> fun log -> log.Trim('[', ']', ':')
    |> fun log -> log.ToLower()

let reformat(logLine: string): string =
    (message logLine, logLevel logLine)
    ||> fun msg log -> $"{msg} ({log})"

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
  std::string message(std::string msg) {
    size_t i = msg.find(": ");
    return msg.substr(i + 2);
  }

  std::string log_level(std::string msg) {
    size_t s = msg.find("[") + 1;
    size_t e = msg.find("]") - 1;
    return msg.substr(s, e);
  }

  std::string reformat(std::string msg) {
    return message(msg) + " (" + log_level(msg) + ")";
  }
} // namespace log_line
